### ISSUE TITLE
IDbConnection.UpdateOnly bug

### ISSUE THREAD

**Issue by labilbe** [ServiceStack/Issues#449]
Using ServiceStack v4.0.60.
An exception is thrown with the following code:
```
abstract class ProductBase : IHasIntId {
        public int Id { get; set; }
        public string ManufacturingMessage { get; set; }
}

class ProductSheet : ProductBase {

}

Db.UpdateOnly(new ProductSheet { ManufacturingMessage = "toto"}, p => p.ManufacturingMessage, p => p.Id == 23);
```

```
Error thrown is 
à System.Linq.Expressions.Compiler.VariableBinder.Reference(ParameterExpression node, VariableStorageKind storage)
   à System.Linq.Expressions.Compiler.VariableBinder.VisitParameter(ParameterExpression node)
   à System.Linq.Expressions.ParameterExpression.Accept(ExpressionVisitor visitor)
   à System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   à System.Linq.Expressions.ExpressionVisitor.VisitMember(MemberExpression node)
   à System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   à System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   à System.Linq.Expressions.Compiler.VariableBinder.VisitUnary(UnaryExpression node)
   à System.Linq.Expressions.UnaryExpression.Accept(ExpressionVisitor visitor)
   à System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   à System.Linq.Expressions.ExpressionVisitor.Visit(ReadOnlyCollection`1 nodes)
   à System.Linq.Expressions.Compiler.VariableBinder.VisitLambda[T](Expression`1 node)
   à System.Linq.Expressions.Expression`1.Accept(ExpressionVisitor visitor)
   à System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   à System.Linq.Expressions.Compiler.VariableBinder.Bind(LambdaExpression lambda)
   à System.Linq.Expressions.Compiler.LambdaCompiler.Compile(LambdaExpression lambda, DebugInfoGenerator debugInfoGenerator)
   à System.Linq.Expressions.Expression`1.Compile()
   à ServiceStack.ExpressionUtil.CachedExpressionCompiler.Compiler`2.<>c__DisplayClass6.<CompileFromFingerprint>b__3(ExpressionFingerprintChain _)
   à System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   à ServiceStack.ExpressionUtil.CachedExpressionCompiler.Compiler`2.CompileFromFingerprint(Expression`1 expr)
   à ServiceStack.ExpressionUtil.CachedExpressionCompiler.Compiler`2.Compile(Expression`1 expr)
   à ServiceStack.CachedExpressionCompiler.Wrap(Expression arg)
   à ServiceStack.CachedExpressionCompiler.Evaluate(Expression arg)
   à ServiceStack.ExpressionUtils.GetFieldNames[T](Expression`1 expr)
   à ServiceStack.OrmLite.SqlExpression`1.Update(Expression`1 fields)
   à ServiceStack.OrmLite.WriteExpressionCommandExtensions.UpdateOnly[T](IDbCommand dbCmd, T obj, Expression`1 onlyFields, Expression`1 where)
   à ServiceStack.OrmLite.OrmLiteWriteExpressionsApi.<>c__DisplayClassa`1.<UpdateOnly>b__9(IDbCommand dbCmd)
   à ServiceStack.OrmLite.OrmLiteExecFilter.Exec[T](IDbConnection dbConn, Func`2 filter)
   à ServiceStack.OrmLite.OrmLiteReadExpressionsApi.Exec[T](IDbConnection dbConn, Func`2 filter)
   à ServiceStack.OrmLite.OrmLiteWriteExpressionsApi.UpdateOnly[T](IDbConnection dbConn, T obj, Expression`1 onlyFields, Expression`1 where)
Result Message:	System.InvalidOperationException : la variable 'p' de type 'Ksm.Gecko.Services.ServiceModel.Types.Common.ProductSheet' est référencée à partir de l'étendue '', mais elle n'est pas définie
```

If I override base properties, no more error is thrown.
So the following modification fixes the problem
```
class ProductSheet : ProductBase {
        public new string ManufacturingMessage { get; set; }
}

```

I don't know if it is a regression or if the behavior is here since the beginning.

**Comment by xplicit** [comment]
This regression was introduced in this commit https://github.com/ServiceStack/ServiceStack.OrmLite/commit/3370cf3210bd2133d8423042ea57c559d8ba5ee9

**Comment by mythz** [comment]
This should now be fixed in latest v4.0.61 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by labilbe** [comment]
Thank you very much for your reactivity!
